lib/SearchAndSort/SearchAndSort.js

```diff
@@ -504,7 +505,12 @@ class SearchAndSort extends React.Component {
       locallyChangedSearchTerm,
       locallyChangedQueryIndex,
     } = this.state;
+    const { validateSearchOnSubmit } = this.props;
     const qindex = locallyChangedQueryIndex || this.props.selectedIndex;
+    const isSearchValid = validateSearchOnSubmit?.(locallyChangedSearchTerm);
+    const shouldPreventSearch = validateSearchOnSubmit && !isSearchValid;
```

Comment:
It's a little awkward to check the presence of `validateSearchOnSubmit` twice here, and since you never use `shouldPreventSearch` again, storing it seems superfluous. Maybe check all the conditions at once?
```
if (validateSearchOnSubmit && !validateSearchOnSubmit(locallyChangedSearchTerm) {
  return;
}
```

Reply:
I was just going to say exactly the same thing here.

Reply:
Looks great! thanks